File: app/src/main/java/com/absinthe/anywhere_/ui/fragment/CardSharingDialogFragment.java
package com.absinthe.anywhere_.ui.fragment;

import android.app.Dialog;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.absinthe.anywhere_.R;
import com.absinthe.anywhere_.utils.ToastUtil;
import com.absinthe.anywhere_.view.AnywhereDialogBuilder;
import com.absinthe.anywhere_.view.AnywhereDialogFragment;
import com.absinthe.anywhere_.viewbuilder.entity.CardSharingBuilder;

public class CardSharingDialogFragment extends AnywhereDialogFragment {
    private Context mContext;
    private String mText;

    public CardSharingDialogFragment(String text) {
        mText = text;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        mContext = getActivity();
    }

    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AnywhereDialogBuilder builder = new AnywhereDialogBuilder(mContext);
        CardSharingBuilder cardSharingBuilder = new CardSharingBuilder(mContext, mText);

        return builder.setView(cardSharingBuilder.getRoot())
                .setTitle(R.string.menu_share_card)
                .setPositiveButton(R.string.dialog_copy, (dialog, which) -> {
                    ClipboardManager cm = (ClipboardManager) mContext.getSystemService(Context.CLIPBOARD_SERVICE);
                    ClipData mClipData = ClipData.newPlainText("Label", mText);
                    if (cm != null) {
                        cm.setPrimaryClip(mClipData);
                        ToastUtil.makeText(R.string.toast_copied);
                    }
                })
                .create();
    }
}

File: app/src/main/java/com/absinthe/anywhere_/ui/fragment/CardSharingDialogFragment.kt
package com.absinthe.anywhere_.ui.fragment

import android.app.Dialog
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import com.absinthe.anywhere_.R
import com.absinthe.anywhere_.utils.ToastUtil
import com.absinthe.anywhere_.view.app.AnywhereDialogBuilder
import com.absinthe.anywhere_.view.app.AnywhereDialogFragment
import com.absinthe.anywhere_.viewbuilder.entity.CardSharingBuilder

class CardSharingDialogFragment(private val mText: String) : AnywhereDialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AnywhereDialogBuilder(requireContext()).setView(CardSharingBuilder(requireActivity(), mText).root)
                .setTitle(R.string.menu_share_card)
                .setPositiveButton(R.string.dialog_copy) { _: DialogInterface?, _: Int ->
                    val cm = requireContext().getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
                    val mClipData = ClipData.newPlainText("Label", mText)

                    cm.setPrimaryClip(mClipData)
                    ToastUtil.makeText(R.string.toast_copied)
                }
                .create()
    }

}